shared/tls/tls.go

```diff
@@ -31,26 +61,105 @@ func RFC3493Dialer(_ context.Context, network string, address string) (net.Conn,
 		return nil, err
 	}
 
+	// Sort addresses with IPv6 first per RFC 8305
+	addrs = sortAddressesByFamily(addrs)
+
+	// If the context doesn't have a deadline, add one
+	if _, hasDeadline := ctx.Deadline(); !hasDeadline {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, happyEyeballsTimeout)
+		defer cancel()
+	}
+
+	type dialResult struct {
+		conn net.Conn
+		err  error
+		addr string
+	}
+
+	results := make(chan dialResult, len(addrs))
+	var pendingDials int
+
+	// Use a dialer with the context for proper cancellation
+	dialer := &net.Dialer{
+		Timeout: 10 * time.Second,
+	}
+
+	// Start connection attempts with staggered delays per RFC 8305
+	for i, addr := range addrs {
+		// Wait for the staggered delay before starting this attempt,
+		// but check if we already have a successful connection
+		if i > 0 {
+			select {
+			case result := <-results:
+				pendingDials--
+				if result.err == nil {
+					return configureConnection(result.conn)
+				}
+				// Put the error result back for later collection
+				results <- result
+				pendingDials++
+			case <-time.After(happyEyeballsDelay):
+				// Timeout elapsed, start next connection attempt
+			case <-ctx.Done():
+				return nil, fmt.Errorf("%s: %s (context cancelled)", connectErrorPrefix, address)
+			}
+		}
+
+		pendingDials++
+		go func(addr string) {
+			target := net.JoinHostPort(addr, port)
+			conn, err := dialer.DialContext(ctx, network, target)
+			results <- dialResult{conn: conn, err: err, addr: addr}
+		}(addr)
+	}
+
+	// Collect results
 	var errs []error
-	for _, a := range addrs {
-		c, err := net.DialTimeout(network, net.JoinHostPort(a, port), 10*time.Second)
-		if err != nil {
-			errs = append(errs, err)
-			continue
+	var connections []net.Conn
+
+	for pendingDials > 0 {
+		select {
+		case result := <-results:
+			pendingDials--
+			if result.err != nil {
+				errs = append(errs, result.err)
+			} else {
+				connections = append(connections, result.conn)
+			}
```

Comment:
Static analysis seems to want a line break here.

Reply:
Should be resolved now.